cmd/zed/zed.d/statechange-led.sh

```diff
@@ -122,54 +139,66 @@ process_pool()
 
 	# The output will be the vdevs only (from "grep '/dev/'"):
 	#
-	#    U45     ONLINE       0     0     0   /dev/sdk          0
-	#    U46     ONLINE       0     0     0   /dev/sdm          0
-	#    U47     ONLINE       0     0     0   /dev/sdn          0
-	#    U50     ONLINE       0     0     0  /dev/sdbn          0
+	#    U45     ONLINE       0     0     0   /dev/sdk          0       0
+	#    U46     ONLINE       0     0     0   /dev/sdm          0       0
+	#    U47     ONLINE       0     0     0   /dev/sdn          0       0
+	#    U50     ONLINE       0     0     0   Potential info message /dev/sdbn          0       0
 	#
-	ZPOOL_SCRIPTS_AS_ROOT=1 $ZPOOL status -c upath,fault_led "$pool" | grep '/dev/' | (
+	ZPOOL_SCRIPTS_AS_ROOT=1 $ZPOOL status -c upath,fault_led,locate_led "$pool" | grep '/dev/' | (
 	rc=0
 	while read -r vdev state _ _ _ therest; do
-		# Read out current LED value and path
+		# Read out current LED values and path
 		# Get dev name (like 'sda')
-		dev=$(basename "$(echo "$therest" | awk '{print $(NF-1)}')")
+		dev=$(basename "$(echo "$therest" | awk '{print $(NF-2)}')")
 		vdev_enc_sysfs_path=$(realpath "/sys/class/block/$dev/device/enclosure_device"*)
-		current_val=$(echo "$therest" | awk '{print $NF}')
-
-		if [ "$current_val" != "0" ] ; then
-			current_val=1
-		fi
-
 		if [ -z "$vdev_enc_sysfs_path" ] ; then
 			# Skip anything with no sysfs LED entries
 			continue
 		fi
 
-		if [ ! -e "$vdev_enc_sysfs_path/fault" ] ; then
-			rc=3
-			zed_log_msg "vdev $vdev '$file/fault' doesn't exist"
-			continue
-		fi
+		fault_val=$(echo "$therest" | awk '{print $(NF-1)}')
+		locate_val=$(echo "$therest" | awk '{print $NF}')
 
-		val=$(state_to_val "$state")
-
-		if [ "$current_val" = "$val" ] ; then
-			# LED is already set correctly
-			continue
+		# Treat all non-zero values as 1
+		if [ "$fault_val" != "0" ] ; then
+			fault_val=1
 		fi
-
-		if ! check_and_set_led "$vdev_enc_sysfs_path/fault" "$val"; then
-			rc=3
+		if [ "$locate_val" != "0" ] ; then
+			locate_val=1
 		fi
+
+		for encled in fault locate; do
+			if [ ! -e "$vdev_enc_sysfs_path/$encled" ] ; then
+				rc=3
+				zed_log_msg "vdev $vdev '$file/$encled' doesn't exist"
+				continue
+			fi
+
+			val=$(state_to_val "$encled" "$state")
+
+			eval "current_val=\$${encled}_val"
+			# SC gets fooled by the eval workaround above.
+			# shellcheck disable=SC2154
+			if [ "$current_val" = "$val" ] ; then
+				# LED is already set correctly
+				continue
+			fi
+
+			if ! check_and_set_led "$vdev_enc_sysfs_path/$encled" "$val"; then
+				rc=3
+			fi
+		done
 	done
 	exit "$rc"; )
 }
 
 if [ -n "$ZEVENT_VDEV_ENC_SYSFS_PATH" ] && [ -n "$ZEVENT_VDEV_STATE_STR" ] ; then
 	# Got a statechange for an individual vdev
-	val=$(state_to_val "$ZEVENT_VDEV_STATE_STR")
 	vdev=$(basename "$ZEVENT_VDEV_PATH")
-	check_and_set_led "$ZEVENT_VDEV_ENC_SYSFS_PATH/fault" "$val"
+	for encled in fault locate ; do
```

Comment:
space before semicolon

Reply:
In a271946d724ae1d399564ace759fce8e00686812